internal/charmstore/store.go

```diff
@@ -788,3 +797,44 @@ func (s *Store) SynchroniseElasticsearch() error {
 	}
 	return nil
 }
+
+//GetPromulgatedURL finds the promulgatedURL that should be used for
+//this newly uploaded charm, if the charm should be promulgated,
+//othewise it returns nil. An error is returned if there is a problem
+//communicating with the storage.
+func (s *Store) GetPromulgatedURL(id *charm.Reference) (*charm.Reference, error) {
```

Comment:
I wasn't sure where to put it and chose on the store as it does mostly store things. I've moved it.